### PR TITLE
Ref task 3029 handle with item

### PR DESCRIPTION
**Pull request by manhphan34**
### Summary:
- [Feature] Profile Screen
### Link Issue:
- https://edu-redmine.framgia.vn/issues/3029

### CODE REVIEW

**Comment by manhphan34** [comment]
ready

**Comment by tuanlvt-1117** [comment]
commented

**Comment by manhphan34** [comment]
ready

File: app/src/main/java/framgia/com/ichat/data/repository/UserRepository.java [hunk 1]
```diff
@@ -35,4 +41,29 @@ public void getUsers(ValueEventListener valueEventListener) {
     public void getUser(FirebaseUser user, ValueEventListener valueEventListener) {
         mRemote.getUser(user, valueEventListener);
     }
+
+    @Override
+    public void uploadImage(FirebaseUser user, Uri file, OnCompleteListener onCompleteListener, OnFailureListener onFailureListener) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/data/repository/UserRepository.java]: > 80 chars

File: app/src/main/java/framgia/com/ichat/data/repository/UserRepository.java [hunk 2]
```diff
@@ -35,4 +41,29 @@ public void getUsers(ValueEventListener valueEventListener) {
...
+        mRemote.uploadImage(user, file, onCompleteListener, onFailureListener);
+    }
+
+    @Override
+    public void uploadImage(FirebaseUser user, ByteArrayOutputStream bytes, OnCompleteListener onCompleteListener, OnFailureListener onFailureListener) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/data/repository/UserRepository.java]: > 80 chars

File: app/src/main/java/framgia/com/ichat/data/source/remote/UserRemoteDataSource.java [hunk 3]
```diff
@@ -1,29 +1,44 @@
...
 
 public class UserRemoteDataSource implements UserDataSouce.Remote {
+    private static final String PNG = ".png";
     private FirebaseDatabase mDatabase;
+    private FirebaseAuth mAuth;
+    private FirebaseStorage mStorage;
     private static UserRemoteDataSource sInstance;
 
-    public static UserDataSouce.Remote getInstance(FirebaseDatabase database) {
+    public static UserDataSouce.Remote getInstance(FirebaseDatabase database, FirebaseStorage storage, FirebaseAuth auth) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/data/source/remote/UserRemoteDataSource.java]: > 80 chars

File: app/src/main/java/framgia/com/ichat/data/repository/UserRepository.java [hunk 4]
```diff
@@ -35,4 +41,29 @@ public void getUsers(ValueEventListener valueEventListener) {
...
+        mRemote.updateUser(user);
+    }
+
+    @Override
+    public void getLinkImage(OnCompleteListener onCompleteListener) {
+        mRemote.getLinkImage(onCompleteListener);
+    }
+
+    @Override
+    public void updateUserName(String uid, String userName) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/framgia/com/ichat/data/repository/UserRepository.java]: rename value from uid to id

File: app/src/main/java/framgia/com/ichat/data/source/UserDataSouce.java [hunk 5]
```diff
@@ -1,12 +1,34 @@
...
+
+        void uploadImage(FirebaseUser user, ByteArrayOutputStream bytes,
+                         OnCompleteListener onCompleteListener,
+                         OnFailureListener onFailureListener);
+
+        void updateUser(User user);
+
+        void getLinkImage(OnCompleteListener onCompleteListener);
+
+        void updateUserName(String uid, String userName);
```
Comment (tuanlvt-1117) [inline app/src/main/java/framgia/com/ichat/data/source/UserDataSouce.java]: as above

File: app/src/main/res/drawable/ic_error.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="#E50C08"
+    android:viewportHeight="24.0" android:viewportWidth="24.0"
```
Comment (tuanlvt-1117) [inline app/src/main/res/drawable/ic_error.xml]: reformat code

**Comment by tuanlvt-1117** [comment]
commented

File: app/src/main/java/framgia/com/ichat/data/source/remote/UserRemoteDataSource.java [hunk 7]
```diff
@@ -38,4 +53,44 @@ public void getUser(FirebaseUser user, ValueEventListener valueEventListener) {
                 .child(user.getUid())
                 .addValueEventListener(valueEventListener);
     }
+
+    @Override
+    public void uploadImage(FirebaseUser user, Uri file, OnCompleteListener onCompleteListener, OnFailureListener onFailureListener) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/data/source/remote/UserRemoteDataSource.java]: > 80 chars

File: app/src/main/java/framgia/com/ichat/data/source/remote/UserRemoteDataSource.java [hunk 8]
```diff
@@ -38,4 +53,44 @@ public void getUser(FirebaseUser user, ValueEventListener valueEventListener) {
...
+        StorageReference storageReference = mStorage.getReference();
+        storageReference.child(user.getUid().concat(PNG))
+                .putFile(file)
+                .addOnCompleteListener(onCompleteListener)
+                .addOnFailureListener(onFailureListener);
+    }
+
+    @Override
+    public void uploadImage(FirebaseUser user, ByteArrayOutputStream bytes, OnCompleteListener onCompleteListener, OnFailureListener onFailureListener) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/data/source/remote/UserRemoteDataSource.java]: > 80 chars

**Comment by NguyenMinhTuanB** [comment]
commented

**Comment by manhphan34** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Comment by tuanlvt-1117** [comment]
merged

**Pull request merged**